pkg/kubelet/cm/cpuset/cpuset.go

```diff
@@ -14,75 +14,48 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+// cpuset represnts a set of CPUs in a 'set' data structure.
+//
+// It can be used to represent core IDs, hyper thread siblings, CPU nodes, or processor IDs.
+//
+// The only special thing about this package is that
+// methods are provided to convert back and forth from Linux 'list' syntax.
+// See http://man7.org/linux/man-pages/man7/cpuset.7.html#FORMATS for details.
+//
+// Future work can migrate this to use a 'set' library, and relax the dubious 'immutable' property.
 package cpuset
 
 import (
 	"bytes"
 	"fmt"
-	"os"
 	"reflect"
 	"sort"
 	"strconv"
 	"strings"
-
-	"k8s.io/klog/v2"
 )
 
-// Builder is a mutable builder for CPUSet. Functions that mutate instances
-// of this type are not thread-safe.
-type Builder struct {
-	result CPUSet
-	done   bool
-}
-
-// NewBuilder returns a mutable CPUSet builder.
-func NewBuilder() *Builder {
-	return &Builder{
-		result: CPUSet{
-			elems: map[int]struct{}{},
-		},
-	}
-}
-
-// Add adds the supplied elements to the result. Calling Add after calling
-// Result has no effect.
-func (b *Builder) Add(elems ...int) {
-	if b.done {
-		return
-	}
-	for _, elem := range elems {
-		b.result.elems[elem] = struct{}{}
-	}
-}
-
-// Result returns the result CPUSet containing all elements that were
-// previously added to this builder. Subsequent calls to Add have no effect.
-func (b *Builder) Result() CPUSet {
-	b.done = true
-	return b.result
-}
-
 // CPUSet is a thread-safe, immutable set-like data structure for CPU IDs.
 type CPUSet struct {
 	elems map[int]struct{}
 }
 
```

Comment:
It's still unclear to me why we don't just implement this in terms of a `sets.Set[int]` from the `apimachinery/pkg/util/sets` package. Most of the functions are already implemented (so would be a simple passthrough), and the one's that aren't can easily be implemented in terms of the others.

I had previously suggested this in https://github.com/kubernetes/utils/pull/267#discussion_r1028565778, but no one responded.

The only reason I can see *not* doing this is if that would prevent us from moving the implementation to `kubernetes/utils`.  I don't know what the requirements are there in terms of package imports.

/cc @thockin

Reply:
Hi Kevin,

First, I agree that is a good goal (it is acknolwedged in the package comment, and has been discussed and not ignored, despite how your comment thread appears)

If you review my issue though, the problem I'm trying to solve is exactly exposing this in k/utils, so, depending on something *not* available in k/utils seems counter productive to the primary goal.

To me, a sensible strategy is:
1) keep current implementation of cpuset
2) move cpuset to k/utils
3) move set to k/utils also
4) reimplement cpuset in terms of set at that point